.github/workflows/merge-by/build-table-and-notify.js

```diff
@@ -0,0 +1,206 @@
+      owner,
+      repo,
+      issue_number: pr.number,
+      body
+    });
+  }
+};
+
+/**
+ * Fetches PRs with a merge-by date < 1 week from now.
+ * 
+ * @param {Object} dayJs Day.js exports for manipulating/querying time differences
+ * @param {Object} github The OctoKit/rest.js API for making requests to GitHub
+ * @param {string} owner The owner of the repository (user or organization)
+ * @param {string} repo The name of the repository on GitHub
+ * @param {Object} today Today's date, represented as a day.js object
+ */
+const fetchPullRequests = async ({ dayJs, github, owner, repo, today }) => {
+  const nextWeek = today.add(7, "day");
+  return (await Promise.all((await github.rest.pulls.list({
+    owner,
+    repo,
+    state: "open"
+  }))?.data.filter((pr) => !pr.draft)
+    .map(async (pr) => {
+      const comments = (await github.rest.issues.listComments({ owner, repo, issue_number: pr.number })).data;
+      // Attempt to parse the merge-by date from the bot comment
+      const existingComment = comments?.find((comment) =>
+        comment.user.login === "github-actions[bot]" && comment.body.includes("**📅 Suggested merge-by date:"));
+
+      const reviews = (await github.rest.pulls.listReviews({
+        owner,
+        repo,
+        pull_number: pr.number,
+      })).data;
+
+      const hasTwoReviews = reviews.reduce((all, review) => review.state === "APPROVED" ? all + 1 : all, 0) >= 2;
+
+      // Filter out reviewers if they have already reviewed and approved the pull request
+      const reviewersNotApproved = pr.requested_reviewers
+        .filter((reviewer) => 
+          reviews.find((review) => review.state === "APPROVED" && reviewer.login === review.user.login) == null);
+
+      return {
+        number: pr.number,
+        title: pr.title,
+        author: pr.user.login,
+        hasReviews: hasTwoReviews,
+        mergeable: pr.mergeable,
+        reviewers: reviewersNotApproved,
+        mergeBy: existingComment?.body.substring(existingComment.body.lastIndexOf("*") + 1).trim()
+      };
+    }))).filter((pr) => {
+      if (pr.mergeBy == null) {
+        return true;
+      }
+
+      // Filter out any PRs that have merge-by dates > 1 week from now
+      const mergeByDate = dayJs(pr.mergeBy);
+      return nextWeek.diff(mergeByDate, "day") <= 7;
+    }).reverse();
+}
+
+module.exports = async ({ github, context }) => {
+  const dayJs = require("dayjs");
+  const today = dayJs();
+  const owner = context.repo.owner;
+  const repo = context.repo.repo;
+  const pullRequests = await fetchPullRequests({ dayJs, github, owner, repo, today });
+  // Look over existing PRs, grab all PRs with a merge-by date <= 1w from now, and update the issue with the new table
+  await scanPRsAndUpdateTable({ github, owner, pullRequests, repo });
+  // Notify users for PRs with merge-by dates coming up within 24hrs from now
+  await notifyUsers({ dayJs, github, owner, pullRequests, repo, today });
```

Comment:
thanks for breaking it down like this. It makes it easier to maintain going forward 😋